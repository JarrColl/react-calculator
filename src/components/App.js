import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

const myData = {
  total: '5',
  next: '3',
  operation: '+',
};

export default function App() {
  calculate(myData, '=');
  return (
    <>
      <Display />
      <ButtonPanel />
    </>
  );
}
