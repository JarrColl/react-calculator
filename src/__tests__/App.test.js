import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import App from '../components/App';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render a Quote', () => {
  act(() => {
    render(<App />, container);
  });

  expect(pretty(container.innerHTML)).toMatchInlineSnapshot(`
    "<div>
      <h3>Let's do some math!</h3>
      <div>
        <div>0</div>
        <div><button type=\\"button\\">AC</button><button type=\\"button\\">+/-</button><button type=\\"button\\">%</button><button type=\\"button\\">รท</button></div>
        <div><button type=\\"button\\">7</button><button type=\\"button\\">8</button><button type=\\"button\\">9</button><button type=\\"button\\">X</button></div>
        <div><button type=\\"button\\">4</button><button type=\\"button\\">5</button><button type=\\"button\\">6</button><button type=\\"button\\">-</button></div>
        <div><button type=\\"button\\">1</button><button type=\\"button\\">2</button><button type=\\"button\\">3</button><button type=\\"button\\">+</button></div>
        <div><button type=\\"button\\">0</button><button type=\\"button\\">.</button><button type=\\"button\\">=</button></div>
      </div>
    </div>"
  `);
});
