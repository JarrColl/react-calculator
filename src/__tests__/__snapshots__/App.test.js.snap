// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3>
    Let's do some math!
  </h3>
  <div>
    <div>
      0
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
